podman/api/client.py

```diff
@@ -378,14 +377,17 @@ def _request(
             APIError: when service returns an error
         """
         # Only set timeout if one is given, lower level APIs will not override None
-        timeout_kw = {}
+        timeout_kw: dict[str, Any] = {}
         timeout = timeout or self.timeout
         if timeout_kw is not None:
             timeout_kw["timeout"] = timeout
 
         compatible = kwargs.get("compatible", False)
         path_prefix = self.compatible_prefix if compatible else self.path_prefix
 
+        if isinstance(path, bytes):
+            path = path.decode()
```

Comment:
@jwhonce Since `path` can be of type `bytes`, converting it to a string is required to get `.lstrip(str)` to be valid. Without conversion, a `TypeError` would be thrown on a bytes path. It seems all calls to `APIClient` are done within the package and all use strings for the `path` parameter. Is so, specifying `path` as `str` only would remove the need for a conversion here.